### DISCUSSION TITLE
Multiple VPS question

### DISCUSSION THREAD

**Discussion by Progressed36** [rspamd/rspamd#3658] [General]
I ahve this theory which i cant seem to get working.

I have 3 mailservers all with their own rspamd.

server 1 is mine
server 2 is a clients
server 3 is a clients

On server 1 i want all the servers visible so I setup a neighbor list. On server 2 and 3 i add a secure IP of my internal IP of server 1.  Server 2 and 3 just runs as is as my clients are able to login to those GUI's. 

Am i missing something or is it not going to work?

**Comment by moisseev** [comment]
> On server 1 i want all the servers visible so I setup a neighbor list.

Also you should setup neighbor lists on 2 and 3. They can contain any entries you want, but you need a `neighbours {}` section in the configuration.

https://rspamd.com/doc/configuration/options.html#neighbours-list

> On server 2 and 3 i add a secure IP of my internal IP of server 1.

It is not necessary. WebUI doesn't require any communication between server backends. All three servers should be just accessible by your web-browser. So all three servers should share the same password or you need to configure secure IP of your web-browser on all three servers.

**Reply by Progressed36** [reply]
I would love a password-less approach when logged in from server 1 (hence the secure_ip approach) and on server 2 and 3 i wanted to give a personal password to the clients themselfs.

**Comment by moisseev** [comment]
> when logged in from server 1

Are you running the web-browser (from where you access WebUI) directly on the server 1?

AFAIR you can configure both: a secure_ip and a password.

**Reply by Progressed36** [reply]
no the webbrowser is not run from server 1. Are you saying the secure_ip has to be my public ip instead of the server ip?

**Comment by moisseev** [comment]
Yes. The `secure_ip` has to be the source IP address of HTTP requests.
If you use a proxy, see https://rspamd.com/doc/faq.html#how-to-use-the-webui-behind-a-proxy-server

**Reply by Progressed36** [reply]
Ok so thats my public IP (which is a dynamic IP) of me as client. But im using a proxy as so:

> <Location /rspamd>
>  Require all granted
>  </Location>
> RewriteEngine On
> RewriteRule ^/spam$ /spam/ [R,L]
>  RewriteRule ^/spam/(.*) http://127.0.0.1:11334/$1 [P]`

How can i pass the host IP from server 1, so i can set that as secure_ip on server 2 and 3?

**Comment by moisseev** [comment]
I've already told you, server 1 IP is not relevant.
I would use VPN or SSH tunnel.

**Reply by mrEckendonk** [reply]
SSH tunnel will give you a conflict with the port 11334 is already in use
